### DISCUSSION TITLE
Questions about "atomic-energy"

### DISCUSSION THREAD

**Discussion by jtyang-chem** [deepmodeling/deepmd-kit#1357] [Q&A] [answered]
I have several questions about "atomic-energy" in `training parameters`
- `set_davg_zero`: "Set the normalization average to zero. ", "normalization average" of what? Why it has a relation with `atom_ener`?
- `atom_ener`: Is it fixed in training or will change with training steps when being specified in "input.json"?
- `start_pref_ae`, "If both start_pref_atom_ener and limit_pref_atom_ener are set to 0, then the `atom_ener ` will be ignored.":
 If I set the `atom_ener` in "input.json", not in file "atom_ener.npy", at the same time start/limit_atom_ener are zeros, does my setting of `atom_ener` still count? What is the difference between settings in "input.json" and in "atom_ener.npy" of "atom_ener"?

**Comment by njzjz** [comment, marked as answer]
If you set `atom_ener` <img src="https://render.githubusercontent.com/render/math?math=E%5Ei_0"> that means the atomic energy in the vacuum, the atomic energy <img src="https://render.githubusercontent.com/render/math?math=E%5Ei">  where atoms are not in the vacuum will be given by
![](https://render.githubusercontent.com/render/math?math=E%5Ei%20%3D%20%5Cmathcal%7BN%7D(%5Cmathcal%7BD%7D%5Ei)%20-%20%5Cmathcal%7BN%7D(%5Cmathbf%7BD%7D%5Ei_0)%20%2B%20E%5Ei_0)
where <img src="https://render.githubusercontent.com/render/math?math=%5Cmathbf%7BD%7D%5Ei_0"> is the descriptor when there is only one atom. <img src="https://render.githubusercontent.com/render/math?math=E%5Ei_0"> is a fixed value. Here, for convenience and performance, we assumes <img src="https://render.githubusercontent.com/render/math?math=%5Cmathbf%7BD%7D%5Ei_0%3D%5Cmathbf%7B0%7D"> by setting `set_davg_zero` to `true` in the program.

When you provide `atom_ener.npy` and add the loss function, it is a way to fit <img src="https://render.githubusercontent.com/render/math?math=E%5Ei">, which is different from <img src="https://render.githubusercontent.com/render/math?math=E%5Ei_0">.

**Reply by jtyang-chem** [reply]
Now I think I understand the equation. And it's different from the one in that "[End to End](https://arxiv.org/abs/1805.09003)" paper.  Is this published?

**Reply by njzjz** [reply]
The feature is a part of work from [this paper](https://doi.org/10.1021/acs.jctc.1c00201).
